magicbyirineuTests/Modules/CardDetail/CardDetailViewControllerSpec.swift

```diff
@@ -33,7 +33,7 @@ class CardDetailViewControllerSpec: QuickSpec{
         
         context("when it is initialized with card as not favorite ") {
             it("should match view with snapshot", closure: {
-                sut.screen.favoriteButton.setAsNotFavorite()
+                sut.screen.favoriteButton.setFavorite(isFavorite: false)
```

Comment:
Apesar de estar na classe de teste ainda continua violando a `Law of demeter`. 

Na minha opinião, deveria ter algo assim:

`screen.setFavorite(isFavorite: Bool)` futuramente se eu quiser mudar a forma de como  é favoritado eu teria que mudar as duas classes. Se eu tenho uma abstração de método dentro da VC eu só mudo em um lugar.